### PR TITLE
Add proxy support when sending requests to sandbox API

### LINKED ISSUE #70: Use proxy configured in settings when executing http/https requests when proxy is enabled
http/https client should be configured to use proxy when proxy is enabled in settings and proxy url is configured.

### PR DESCRIPTION
**Pull request by dgolovin**
Fix #70.

Add proxy support for http/https requests to sandbox API.

### CODE REVIEW

File: src/certificates.ts [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+import { isLinux, isMac, isWindows } from './util';
+import { spawnWithPromise } from './util';
+import * as https from 'node:https';
+import * as tls from 'node:tls';
+import * as fs from 'node:fs';
+import * as path from 'node:path';
+import wincaAPI from 'win-ca/api';
+
+/**
+ * Provides access to the certificates of the underlying platform.
```
Comment (benoitf) [inline src/certificates.ts:28]: It looks a duplicate from podman desktop ?
Probably need to expose them from podman desktop or provide http connection ?

Comment (dgolovin) [reply src/certificates.ts:28]: That is correct, at first it looked like small copy/paste, then it got bigger.
The proxy support for extensions should be ether transparent as it is in vscode or there should be http client exposed for extensions to use.

Comment (benoitf) [reply src/certificates.ts:28]: transparent is looking more interesting probably

Comment (dgolovin) [reply src/certificates.ts:28]: Found where it is done for vscode 
https://vscode.dev/github/microsoft/vscode/blob/main/src/vs/workbench/api/node/proxyResolver.ts#L59

**Pull request closed without merging**